superset/sql_parse.py

```diff
@@ -57,10 +58,30 @@ def _extract_limit_from_query(statement: TokenList) -> Optional[int]:
     return None
 
 
+@dataclass(eq=True, frozen=True)
+class Table:  # pylint: disable=too-few-public-methods
+    """
+    A fully specified SQL table conforming to [[catalog.]schema.]table.
+    """
+
+    table: str
+    schema: Optional[str] = None
+    catalog: Optional[str] = None
+
+    def __str__(self) -> str:
```

Comment:
Is there some reason this isn't `__repr__`?

Reply:
@villebro I originally had this as `__repr__` however when using the CLI I find the following more intuative, 

```pthon
In [1]: from superset.sql_parse import Table 
In [2]: table = Table("foo", "bar", "baz")                                                                                                
In [3]: table                                                                                                                             
Out[3]: Table(table='foo', schema='bar', catalog='baz')
In [4]: str(table)                                                                                                                        
Out[4]: 'baz.bar.foo'
```

i.e., it's clearer what the type of `table` is in (3) as opposed to (4).

Reply:
I agree, I didn't realize `dataclass` adds such an intuitive `__repr__`